test/JDBC/expected/temp_table.out

```diff
@@ -63,6 +64,176 @@ int#!#varchar#!#int
 ~~END~~
 
 
+-- BABEL-5273 ALTER COLUMN to another type
+ALTER TABLE #t1 ALTER COLUMN b CHAR(5)
+GO
+
+INSERT INTO #t1 (b) VALUES ('hello')
+GO
+~~ROW COUNT: 1~~
+
+
+SELECT * FROM #t1
+GO
+~~START~~
+int#!#char#!#int
+1#!#<NULL>#!#2
+2#!#hello#!#3
+~~END~~
+
+
+-- should fail due to possible truncation
+ALTER TABLE #t1 ALTER COLUMN b CHAR(4)
+GO
+~~ERROR (Code: 8152)~~
+
+~~ERROR (Message: value too long for type character(4))~~
+
+
+DELETE FROM #t1 WHERE b = 'hello'
+GO
+~~ROW COUNT: 1~~
+
+
+-- try all other TSQL types
+ALTER TABLE #t1 ALTER COLUMN b TINYINT
+GO
+
+ALTER TABLE #t1 ALTER COLUMN b SMALLINT
+GO
+
+ALTER TABLE #t1 ALTER COLUMN b INT
+GO
+
+ALTER TABLE #t1 ALTER COLUMN b BIGINT
+GO
+
+ALTER TABLE #t1 ALTER COLUMN b BIT
+GO
+
+ALTER TABLE #t1 ALTER COLUMN b DECIMAL
+GO
+
+ALTER TABLE #t1 ALTER COLUMN b NUMERIC
+GO
+
+ALTER TABLE #t1 ALTER COLUMN b MONEY
+GO
+
+ALTER TABLE #t1 ALTER COLUMN b SMALLMONEY
+GO
+
+ALTER TABLE #t1 ALTER COLUMN b FLOAT
+GO
+
+ALTER TABLE #t1 ALTER COLUMN b REAL
+GO
+
+-- should raise error due to incompatible types
+ALTER TABLE #t1 ALTER COLUMN b DATE
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: column "b" cannot be cast automatically to type date)~~
+
+
+-- should raise error due to incompatible types
+ALTER TABLE #t1 ALTER COLUMN b TIME
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: column "b" cannot be cast automatically to type time without time zone)~~
+
+
+-- should raise error due to incompatible types
+ALTER TABLE #t1 ALTER COLUMN b DATETIME2
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: column "b" cannot be cast automatically to type datetime2)~~
+
+
+-- should raise error due to incompatible types
+ALTER TABLE #t1 ALTER COLUMN b DATETIMEOFFSET
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: column "b" cannot be cast automatically to type datetimeoffset)~~
+
+
+ALTER TABLE #t1 ALTER COLUMN b DATETIME
+GO
+
+ALTER TABLE #t1 ALTER COLUMN b SMALLDATETIME
+GO
+
+ALTER TABLE #t1 ALTER COLUMN b CHAR
+GO
+
+ALTER TABLE #t1 ALTER COLUMN b VARCHAR
+GO
+
+-- TODO: fix this, it should work and not raise a syntax error
```

Comment:
should this be filed in a separate JIRA? is this temp table specific or just general issue?

Reply:
I believe this is a general issue with altering table datatypes. I'll file a JIRA for this.